visualization/app/codeCharta/util/algorithm/treeMapLayout/treeMapHelper.ts

```diff
@@ -243,6 +247,33 @@ function calculateTrueGradient(mapColors: MapColors, colorRange: ColorRange, met
 	return ColorConverter.convertColorToHex(new Color().lerpColors(neutralColorRGB, negativeColorRGB, negativeFactor))
 }
 
+function calculateFocusedGradient(mapColors: MapColors, colorRange: ColorRange, metricValue: number) {
```

Comment:
We could extract the 3 or 4 calculation methods to separate calculation classes. This would reduce the cyclomatic complexity of treeMapHelper.ts.